From 946380bef36eea875de48ab3de5b13f15b19cc36 Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Thu, 18 Aug 2022 17:38:54 +0530
Subject: [PATCH] ath11k: prevent monitor interface to transmit packet

Firmware crash has observed when monitor interface queues tx
packet. Since firmware does not support tx through monitor interface,
Avoid the packet queuing in drvier itself.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/mac.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/wireless/ath/ath11k/mac.c b/drivers/net/wireless/ath/ath11k/mac.c
index a12c895..38b82be 100644
--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -7422,6 +7422,11 @@ static void ath11k_mac_op_tx(struct ieee80211_hw *hw,
 	int ret;
 	u64 adjusted_tsf;
 
+	if (arvif->vdev_type == WMI_VDEV_TYPE_MONITOR) {
+		ieee80211_free_txskb(ar->hw, skb);
+		return;
+	}
+
 	if (unlikely(test_bit(ATH11K_FLAG_CRASH_FLUSH, &ar->ab->dev_flags))) {
 		ieee80211_free_txskb(ar->hw, skb);
 		return;
-- 
2.7.4

